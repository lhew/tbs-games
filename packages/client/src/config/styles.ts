export enum Screen {
  TABLET = 768,
  TABLET_PORTRAIT = 1024,
  DESKTOP = 1200,
  DESKTOP_LG = 1440,
  DESKTOP_XL = 1600,
}
