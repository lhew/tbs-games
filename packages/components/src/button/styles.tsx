import styled from 'styled-components';

export const Btn = styled.span`
  background: '#DBDBDB';
  padding: 0.7em;
  height: 2em;
`;
