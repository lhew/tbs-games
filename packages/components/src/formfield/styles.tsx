import styled from 'styled-components';
// const tokens = require('@tbs-games/design-tokens');

export const StyledFormField = styled.div`
  display: grid;
  grid-template: ${({ align = 'rows' }: { align?: 'rows' | 'columns' }) =>
    align === 'rows' ? 'auto / 1fr 1fr' : '1fr 1fr / auto'};

  gap: 1em;
  margin-bottom: 1em;
`;

export const StyledLabel = styled.label`
  padding: 0.75em 0;
`;

export const Message = styled.div`
  grid-column: 2;
  font-size: 85%;
`;
