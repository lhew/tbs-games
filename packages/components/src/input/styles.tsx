import styled from 'styled-components';
const tokens = require('@tbs-games/design-tokens');

export const StyledInput = styled.input`
  font-size: 1em;
  background: ${tokens.InputDefaultBackgroundColor};
  color: ${tokens.InputDefaultColor};
  font-family: ${tokens.InputDefaultFontFamily};
  border-radius: ${tokens.ButtonDefaultBorderRadius};
  border: ${tokens.InputDefaultBorderThickness}px solid ${tokens.InputDefaultBorderColor};
  line-height: 1;
  padding: 0.75em;
  height: 2.5em;

  &:focus,
  &:active {
    outline: none;
    color: ${tokens.InputFocusColor};
    border-color: ${tokens.InputFocusBorderColor};
  }

  &:disabled {
    background: ${tokens.InputDisabledBackgroundColor};
    color: ${tokens.InputDisabledColor};
  }
`;
