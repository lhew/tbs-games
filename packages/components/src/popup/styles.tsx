import styled from 'styled-components';
const tokens = require('@tbs-games/design-tokens');

export const StyledPopup = styled.div`
  transition: all 0.3s;
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate3d(-50%, -50%, 0);
  padding: 1.5em;
  background: #fff;
  border-radius: ${tokens.BorderRadiusDefault};
  z-index: 101;
  box-shadow: ${tokens.ShadowDefault};
`;

export const Backdrop = styled.div`
  position: fixed;
  width: 100%;
  height: 100%;
  z-index: 100;
  background: ${({ transparent = true }: { transparent?: boolean }) =>
    transparent ? '#00000000' : `${tokens.ColorPrimary300}50`};
`;
