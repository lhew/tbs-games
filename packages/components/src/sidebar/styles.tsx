import styled from 'styled-components';
const tokens = require('@tbs-games/design-tokens');

export const StyledSidebar = styled.div`
  transition: all 0.3s;
  box-shadow: ${tokens.ShadowDefault};
  padding: 1.5em;
  position: fixed;
  left: ${({ isOpen = true }: { isOpen?: boolean }) => (isOpen ? '0' : '-16em')};
  top: 0;
  height: 100%;
  z-index: 90;
  width: 16em;
`;

export const SidebarBackdrop = styled.div`
  position: fixed;
  width: 100%;
  height: 100%;
  z-index: 91;
  background: ${({ transparent = true }: { transparent?: boolean }) =>
    transparent ? '#00000000' : `${tokens.ColorPrimary300}50`};
`;
