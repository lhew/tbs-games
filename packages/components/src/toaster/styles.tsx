import styled from 'styled-components';
const tokens = require('@tbs-games/design-tokens');

export const StyledToaster = styled.div`
  transition: all 0.3s;
  position: absolute;
  left: 50%;
  bottom: 2em;
  opacity: 0;
  transform: translate3d(0, -50%, 0);
  font-size: 1em;
  text-align: center;
  padding: 1.5em;
  background: ${tokens.ToasterDefaultBackgroundColor};
  color: ${tokens.ToasterDefaultColor};
  font-family: ${tokens.ToasterDefaultFontFamily};
  font-weight: bold;
  border-radius: ${tokens.ToasterDefaultBorderRadius};

  ${({ isOpen = false }: { isOpen?: boolean }) => {
    if (isOpen) {
      return `
        bottom: 1em;
        opacity: 1;
      `;
    }

    return `
      bottom: -2em;
      opacity: 0;
    `;
  }}
`;
