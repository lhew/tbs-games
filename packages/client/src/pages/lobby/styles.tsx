import styled from 'styled-components';
import { Screen } from '../../config/styles';
const tokens = require('@tbs-games/design-tokens');

export const Canvas = styled.div`
  text-align: center;
  height: 100vh;
  display: flex;
  justify-content: center;
  flex-direction: column;
`;

console.log('tokens: ', tokens);

export const StyledButton = styled.span`
  font-size: 1.5em;
  padding: 1.5em;
  text-align: center;
  margin-bottom: 1em;
  cursor: pointer;
  background: ${({ type = 'primary' }: { type?: 'primary' | 'secondary' }) =>
    type === 'primary' ? tokens.ColorPrimary300 : tokens.ColorSecondary300};
  border-radius: ${tokens.BorderRadiusDefault};
  border: none;
  color: #fff;
  &:focus,
  &:active {
    background: ${({ type = 'primary' }: { type?: 'primary' | 'secondary' }) =>
      type === 'primary' ? tokens.ColorPrimary100 : tokens.ColorSecondary100};
  }
`;

export const GameOptionsWrapper = styled.div`
  margin: 2em auto;
  display: grid;
  grid-template: auto auto / auto;
  gap: 1.5em;
  max-width: 25em;
  width: 100%;
  @media screen and (min-width: ${Screen.TABLET}px) {
    grid-template: auto auto / auto;
  }
`;
