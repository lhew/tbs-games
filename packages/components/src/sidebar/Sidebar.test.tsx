import React from 'react';
import TestRenderer from 'react-test-renderer'; // ES6
import Sidebar from '.';
import { StyledSidebar, SidebarBackdrop } from './styles';

test('Sidebar scenarios', () => {
  const defaultSidebar = TestRenderer.create(<Sidebar isOpen>Children item</Sidebar>);
  expect(defaultSidebar).toMatchSnapshot();

  const closedSidebar = TestRenderer.create(<Sidebar>Children item</Sidebar>);
  expect(closedSidebar).toMatchSnapshot();

  // Styled component testing
  const styledSidebar = TestRenderer.create(<StyledSidebar isOpen>Children item</StyledSidebar>);
  expect(styledSidebar).toMatchSnapshot();

  const closedStyledSidebar = TestRenderer.create(<StyledSidebar>Children item</StyledSidebar>);
  expect(closedStyledSidebar).toMatchSnapshot();

  //backdrop

  const opaqueBackdrop = TestRenderer.create(<SidebarBackdrop transparent={false}>Children item</SidebarBackdrop>);
  expect(opaqueBackdrop).toMatchSnapshot();

  const transparentBackdrop = TestRenderer.create(<SidebarBackdrop transparent>Children item</SidebarBackdrop>);
  expect(transparentBackdrop).toMatchSnapshot();

  const defaultBackdrop = TestRenderer.create(<SidebarBackdrop>Children item</SidebarBackdrop>);
  expect(defaultBackdrop).toMatchSnapshot();
});
