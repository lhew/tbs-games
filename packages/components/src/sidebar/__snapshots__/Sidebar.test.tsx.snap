// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar scenarios 1`] = `
<div
  className="sc-AxmLO gIELgM"
>
  Children item
</div>
`;

exports[`Sidebar scenarios 2`] = `
<div
  className="sc-AxmLO jFAXou"
>
  Children item
</div>
`;

exports[`Sidebar scenarios 3`] = `
<div
  className="sc-AxmLO gIELgM"
>
  Children item
</div>
`;

exports[`Sidebar scenarios 4`] = `
<div
  className="sc-AxmLO gIELgM"
>
  Children item
</div>
`;

exports[`Sidebar scenarios 5`] = `
<div
  className="sc-fzozJi jxDyMo"
>
  Children item
</div>
`;

exports[`Sidebar scenarios 6`] = `
<div
  className="sc-fzozJi joanvg"
>
  Children item
</div>
`;

exports[`Sidebar scenarios 7`] = `
<div
  className="sc-fzozJi joanvg"
>
  Children item
</div>
`;
